comp/dogstatsd/listeners/udp_test.go

```diff
@@ -100,26 +98,15 @@ func TestUDPListenerTelemetry(t *testing.T) {
 		telemetryMock, ok := deps.Telemetry.(telemetry.Mock)
 		assert.True(t, ok)
 
-		registry := telemetryMock.GetRegistry()
-		var packetsMetric []*dto.Metric
-		var bytesCountMetric []*dto.Metric
-		metricsFamily, err := registry.Gather()
-		assert.Nil(t, err)
-
-		for _, metric := range metricsFamily {
-			if metric.GetName() == "dogstatsd__udp_packets" {
-				packetsMetric = metric.GetMetric()
-			}
-			if metric.GetName() == "dogstatsd__udp_packets_bytes" {
-				bytesCountMetric = metric.GetMetric()
-			}
-		}
-
-		assert.NotNil(t, packetsMetric)
-		assert.NotNil(t, bytesCountMetric)
+		packetsMetrics, err := telemetryMock.GetCountMetric("dogstatsd", "udp_packets")
```

Comment:
I think all cases below could be distilled down to one line:

```suggestion
		telemetryMock.AssertCountMetric(t, "dogstatsd", "udp_packets", []string{}, 1)
```

But what you have also works.

Reply:
That is a great feedback. I would like to see how the adoption of the helpers goes, and if we identify that most of the usage is similar we can improve the helper functions 😄